feature/policy_forwarding/GUE/Outer_IPv4/decapsulation/otg_tests/Inner_IP/README.md

```diff
@@ -0,0 +1,337 @@
+
+## Testbed type
+
+* [`featureprofiles/topologies/atedut_2.testbed`](https://github.com/openconfig/featureprofiles/blob/main/topologies/atedut_2.testbed)
+
+## Procedure
+
+### Test environment setup
+
+* Create the following connections:
+* DUT has ingress and egress port connected to the ATE.
+  
+```mermaid
+graph LR; 
+A[ATE:Port1] --Ingress--> B[Port1:DUT:Port2];B --Egress--> C[Port2:ATE];
+```
+
+*  ATE Port 1: Generates GUE-encapsulated traffic with various inner (original) destinations.
+*  ATE Port 2: Receives decapsulated traffic whose inner destination matches the policy.
+  
+### DUT Configuration
+
+1.  Interfaces: Configure all DUT ports as singleton IP interfaces.
+ 
+2.  Static Routes/LSPs:
+    *  Configure an IPv4 static route to GUE decapsulation destination (DECAP-DST) to Null0.
+    *  Have policy configuration that match GUE decapsulation destination and default/non-default GUE UDP port/port-range for the decapsulation.
+       *  If udp port is not configured then the default GUE UDP port will be inherited/used.
+    *  Apply the defined policy on the Ingress (DUT port1) port.
+    *  Configure static routes for encapsulated traffic destinations IPV4-DST1 and IPV6-DST1 towards ATE Port 2.
+    *  Configure static MPLS label binding (LBL1) towards ATE Port 2. Next hop of ATE Port 1 should be indicated for MPLS pop action.
+    *  Configure static routes for destination IPV4-DST2 and IPV6-DST2 towards ATE Port 2.
+
+3.  Policy-Based Forwarding: 
+    *  Rule 1: Match GUE traffic with destination DECAP-DST using destination-address-prefix-set and default/non-default GUE UDP port/port-range for decapsulation.
+      * If udp port is not configured then the default GUE UDP port will be inherited/used.   
+    *  Rule 2: Match all other traffic and forward (no decapsulation).
+    *  Apply the defined policy on the Ingress (DUT port1) port. 
+    
+### PF-1.4.1: GUE Decapsulation of inner IPv4 traffic using default GUE UDP port 6080
+-  Push DUT configuration.
+
+Traffic: 
+-  Generate GUE-encapsulated traffic from ATE Port 1 with destinations matching DECAP-DST and default GUE UDP port 6080.
+-  Inner IPv4 destination should match IPV4-DST1.
+-  Inner-packet DSCP value should be set to 32.
+-  Inner-packet TTL value should be set to 64.
+  
+Verification: 
+-  Decapsulated IPv4 traffic is received on ATE Port 2.
+-  No packet loss.
+-  Inner-packet DSCP should be preserved.
+-  Inner-packet TTL value should be decremented by 1 to 63.
+-  PF counters reflect decapsulated packets.
+
+### PF-1.4.2: GUE Decapsulation of inner IPv4 traffic using non-default GUE UDP port or port-range
+-  Push DUT configuration.
+
+Traffic: 
+-  Generate GUE-encapsulated traffic from ATE Port 1 with destinations matching DECAP-DST and configured non-default GUE UDP port or port-range. 
+-  Inner IPv4 destination should match IPV4-DST1.
+-  Inner-packet DSCP value should be set to 32.
+-  Inner-packet TTL value should be set to 64.
+
+Verification: 
```

Comment:
Suggest considering the suggestions above for verification steps plus expectations.

Reply:
the response is same as above for validation and dscp.